src/arviz_plots/backend/matplotlib/__init__.py

```diff
@@ -225,9 +226,23 @@ def hist(
             facecolor = color
         if edgecolor is unset:
             edgecolor = color
-    kwargs = {"bottom": bottom, "color": facecolor, "edgecolor": edgecolor, "alpha": alpha}
+    kwargs = {"color": facecolor, "alpha": alpha}
+    if step_hist:
+        return target.step(
+            np.r_[l_e, r_e[-1]],
+            np.r_[y, y[-1]],
+            where="post",
+            **_filter_kwargs(kwargs, None, artist_kws),
+        )
+
     return target.bar(
-        l_e, height, width=widths, align="edge", **_filter_kwargs(kwargs, None, artist_kws)
+        l_e,
+        height,
+        width=widths,
+        align="edge",
+        bottom=bottom,
+        edgecolor=edgecolor,
+        **_filter_kwargs(kwargs, None, artist_kws),
```

Comment:
```suggestion
 l_e, height, width=widths, align="edge", **_filter_kwargs(kwargs, None, artist_kws)
```

not clear why are you changing

Reply:
@aloctavodia 

Actually the bottom and edgecolor attributes are not there in step histogram, so if we feel kwargs (containing bottom and edgecolor values) in to step function then it throws error. 
So I removed bottom and edgecolor attributes from kwargs dictionary and I did feed these attributes separately into bar histogram function.

I can create two separate kwargs too for step and bar histograms. Let me know if you want that

Reply:
We should then ignore them